src/control.cc

```diff
@@ -166,7 +166,7 @@ AbstractNode* ControlModule::getChild(const ValuePtr &value, const EvalContext*
 	return modulectx->getChild(n)->evaluate(modulectx);
 }
 
-AbstractNode *ControlModule::instantiate(const Context* ctx, const ModuleInstantiation *inst, EvalContext *evalctx) const
+AbstractNode *ControlModule::instantiate(const std::shared_ptr<Context> ctx, const ModuleInstantiation *inst, std::shared_ptr<EvalContext> evalctx) const
```

Comment:
I _think_ there is a small optimization potential in passing a reference to the `shared_ptr` instances, as it may not have to increase the reference count for every call.

Reply:
Done. Micro benchmark with recursive fibonacci shows small improvement but in total still quite a bit slower than the nightly build.
```
function fib(x) = x < 2 ? x : fib(x - 1) + fib(x - 2);

28f9df23 - Only use forward declaration for Expression.
36: 1m31,538s
40: 10m35,664s

0aac8bed - Pass context via reference.
36: 1m23,405s
40: 9m36,937s

Nightly
36: 0m36,848s
40: 4m22,674s
```

Reply:
With the latest changes, this is now at `36: 1m2,891s`.

Reply:
With the `Value` rework changes, results are `36: 0m51,910s`.